app/shared/requiresSignin.ts

```diff
@@ -1,9 +1,19 @@
 import * as pathLib from "path";
+import parse from "url-parse";
 
 // To avoid security vulnerabilities do not add public paths that do not end in a slash
-const publicPaths = ["/api/contact-us/", "/help-centre/", "/api/known-issues/"];
+const publicPaths = [
+  "/api/contact-us/",
+  "/help-centre/",
+  "/api/known-issues/",
+  "/cancel-reminders/",
+  "/api/reminders/cancel/"
+];
 
 export const requiresSignin = (path: string) => {
-  const normalizedPath = pathLib.normalize(path + "/");
+  const parsed = parse(path);
+  const normalizedPath = pathLib.normalize(
+    parsed.pathname + "/" + parsed.query
+  );
```

Comment:
I don't understand the benefit of this change. Could you clarify?

Reply:
our new endpoint requires a querystring, this change makes that possible

Reply:
We are using `startsWith` in the next line. Wouldn't that work without changes?

Reply:
I think the old code just stuck a `/` at the end of the querystring

Reply:
yeah I took a look at the old code just now - it was adding `/` after the query string which was then not being matched, so it would redirect to the sign in page

Reply:
`'/cancel-reminders/<querystringparam>/'.startsWith('/cancel-reminders/')` would still return true as desired though. No?

Reply:
I just got it!
You were using query string params, which mean the comparison would actually be `'/cancel-reminders?<querystringparam>/'.startsWith('/cancel-reminders/')`, not what I posted on the previous comment.